src/StreamSelectLoop.php

```diff
@@ -196,6 +237,9 @@ private function waitForStreamActivity($timeout)
         $write = $this->writeStreams;
 
         $available = $this->streamSelect($read, $write, $timeout);
+        if ($this->pcntl) {
+            \pcntl_signal_dispatch();
+        }
```

Comment:
Signals aren't picked up when I change `if ($this->pcntl) {` into `if (false === $available && $this->pcntl) {`only dispatching when the loop received a signal. Not aware of any performance degradations.